nodebuilder/p2p/p2p.go

```diff
@@ -0,0 +1,213 @@
+	// Connectedness returns a state signaling connection capabilities.
+	Connectedness(id peer.ID) network.Connectedness
+	// NATStatus returns the current NAT status.
+	NATStatus() network.Reachability
+
+	// BlockPeer adds a peer to the set of blocked peers. // TODO should we wrap BlockPeer so
+	//  that it 1. disconnects from peer, then 2. adds to blocklist? cc @Wondertan
+	BlockPeer(p peer.ID) error
+	// UnblockPeer removes a peer from the set of blocked peers.
+	UnblockPeer(p peer.ID) error
+	// ListBlockedPeers returns a list of blocked peers.
+	ListBlockedPeers() []peer.ID
+	// MutualAdd adds a peer to the list of peers who have a bidirectional
+	// peering agreement that they are protected from being trimmed, dropped
+	// or negatively scored.
+	MutualAdd(id peer.ID, tag string)
+	// MutualRm removes a peer from the list of peers who have a bidirectional
+	// peering agreement that they are protected from being trimmed, dropped
+	// or negatively scored, returning a bool representing whether the given
+	// peer is protected or not.
+	MutualRm(id peer.ID, tag string) bool
+	// IsMutual returns whether the given peer is a mutual peer.
+	IsMutual(id peer.ID, tag string) bool
+
+	// BandwidthStats returns a Stats struct with bandwidth metrics for all
+	// data sent/received by the local peer, regardless of protocol or remote
+	// peer IDs.
+	BandwidthStats() metrics.Stats
+	// BandwidthForPeer returns a Stats struct with bandwidth metrics associated with the given peer.ID.
+	// The metrics returned include all traffic sent / received for the peer, regardless of protocol.
+	BandwidthForPeer(id peer.ID) metrics.Stats
+	// BandwidthForProtocol returns a Stats struct with bandwidth metrics associated with the given protocol.ID.
+	BandwidthForProtocol(proto protocol.ID) metrics.Stats
+
+	// ResourceState returns the state of the resource manager.
+	ResourceState() (rcmgr.ResourceManagerStat, error)
+
+	// PubSubPeers returns the peer IDs of the peers joined on
+	// the given topic.
+	PubSubPeers(topic string) []peer.ID
+}
+
+// manager contains all components necessary to access information and
+// perform actions related to the node's p2p Host / operations.
+type manager struct { // TODO @renaynay: rename ?
+	host      HostBase
+	ps        *pubsub.PubSub
+	nat       autonat.AutoNAT
+	connGater *conngater.BasicConnectionGater
+	bw        *metrics.BandwidthCounter
+	rm        network.ResourceManager
+}
+
+func newManager(
+	host host.Host,
+	ps *pubsub.PubSub,
+	nat autonat.AutoNAT,
+	cg *conngater.BasicConnectionGater,
+	bw *metrics.BandwidthCounter,
+	rm network.ResourceManager,
+) Module {
+	return &manager{
+		host:      host,
+		ps:        ps,
+		nat:       nat,
+		connGater: cg,
+		bw:        bw,
+		rm:        rm,
+	}
+}
+
+// Info contains basic information about the node's p2p host/operations.
+type Info struct {
+	// ID is the node's peer ID
+	ID peer.ID `json:"id"`
+	// Addrs is the node's
+	Addrs []ma.Multiaddr `json:"addrs"`
+}
```

Comment:
Not really sure what else is important to return @Wondertan

Reply:
we can add more if we think of any. would merge with this as is

Reply:
agreed